src/storage/src/hummock/vector/writer/mod.rs

```diff
@@ -114,9 +113,10 @@ impl VectorWriterImpl {
         }))
     }
 
-    #[expect(dead_code)]
     pub(crate) async fn flush(&mut self) -> HummockResult<usize> {
-        if let Some((file_info, _blocks, _meta)) = self.vector_file_builder.finish().await? {
+        if let Some((file_info, blocks, meta)) = self.vector_file_builder.finish().await? {
```

Comment:
It seems that the flush for vector data is synchornous. Will it blocks streaming or the `flush` method is called on barrier collection instead of inside the streaming executor?

Reply:
This is by design for simplicity on initial implementation. In this future we can further optimize it to asynchronous flush on uploader.